pyproject.toml

```diff
@@ -84,3 +84,49 @@ reportUnnecessaryTypeIgnoreComment = true
 # Make sure we're checking against the minimum supported Python version by
 # default. CI will check against all supported versions for us.
 pythonVersion = "3.8"
+
+
+[tool.ruff]
```

Comment:
question: why move this from `ruff.toml`? 
`pyproject.toml` is included the built package, so the move will add a few bytes to the overall package size

Reply:
I was having an issue with Ruff not picking up the minimum Python version (3.8). [This comment](https://github.com/astral-sh/ruff/issues/10299#issuecomment-1986575383) suggests that if there's a `ruff.toml`, it won't check the `pyproject.toml` for `project.requires-python`, so rather than duplicate the Python version in the Ruff config, I opted to move it all here.

Is that reason enough, do you think?

Reply:
definitely. also, TIL!

Reply:
For a bit more context, Ruff was breaking the substitution expressions in some f-strings over several lines, which only works in 3.12 onwards.

```python
# Fine in >=3.8
f"This works: {foo}"

# Only fine in >=3.12
f"This causes issues: {
    foo
}"
```